qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,312 @@
+
+        # Solve the relaxed problem
+        start_time_relaxed = timeit.default_timer()
+        relaxed_result = self.min_eigen_solver.compute_minimum_eigenvalue(
+            encoding.qubit_op, aux_operators=variable_ops
+        )
+        relaxed_result.time_taken = timeit.default_timer() - start_time_relaxed
+
+        # Get auxiliary expectation values for rounding.
+        if relaxed_result.aux_operators_evaluated is not None:
+            expectation_values = [v[0] for v in relaxed_result.aux_operators_evaluated]
+        else:
+            expectation_values = None
+
+        # Get the circuit corresponding to the relaxed solution.
+        if isinstance(relaxed_result, VariationalResult):
+            circuit = relaxed_result.optimal_circuit.bind_parameters(relaxed_result.optimal_point)
+        elif isinstance(self.min_eigen_solver, NumPyMinimumEigensolver):
+            statevector = relaxed_result.eigenstate
+            circuit = QuantumCircuit(encoding.num_qubits)
+            circuit.initialize(statevector)
+        else:
+            circuit = None
+
+        rounding_context = RoundingContext(
+            encoding=encoding,
+            expectation_values=expectation_values,
+            circuit=circuit,
+        )
+
+        return relaxed_result, rounding_context
+
+    def solve(self, problem: QuadraticProgram) -> QuantumRandomAccessOptimizationResult:
+        """Solve the relaxed Hamiltonian given by the encoding and round the solution by the given
+            rounding scheme.
+
+        Args:
+            problem: The ``QuadraticProgram`` to be solved.
+
+        Returns:
+            The result of the quantum random access optimization.
+
+        Raises:
+            ValueError: If the encoding has not been encoded with a ``QuadraticProgram``.
+        """
+        # Convert the problem to a QUBO
+        self._verify_compatibility(problem)
+        qubo = self._convert(problem, self._converters)
+        # Encode the QUBO into a quantum random access encoding
+        encoding = QuantumRandomAccessEncoding(max_vars_per_qubit=self.max_vars_per_qubit)
+        encoding.encode(qubo)
+
+        # Solve the relaxed problem
+        relaxed_result, rounding_context = self.solve_relaxed(encoding)
+
+        # Round the solution
+        rounding_result = self.rounding_scheme.round(rounding_context)
+
+        return self.process_result(problem, encoding, relaxed_result, rounding_result)
+
+    def process_result(
+        self,
+        problem: QuadraticProgram,
+        encoding: QuantumRandomAccessEncoding,
+        relaxed_result: MinimumEigensolverResult,
+        rounding_result: RoundingResult,
+    ) -> QuantumRandomAccessOptimizationResult:
+        """Process the relaxed result of the minimum eigensolver and rounding scheme.
+
+        Args:
+            problem: The ``QuadraticProgram`` to be solved.
+            encoding: The ``QuantumRandomAccessEncoding``, for which ``encode()`` must have already
```

Comment:
Maybe we can have similar here for encode statement in terms of method reference as mentioned for solve_relaxed above.

Reply:
updated in the same way as solve_relaxed.